ledger-tool/src/main.rs

```diff
@@ -899,6 +899,80 @@ fn open_blockstore(
     }
 }
 
+fn raw_key_to_slot(key: &[u8], column_name: &str) -> Option<Slot> {
+    match column_name {
+        cf::SlotMeta::NAME => Some(cf::SlotMeta::slot(cf::SlotMeta::index(key))),
+        cf::Orphans::NAME => Some(cf::Orphans::slot(cf::Orphans::index(key))),
+        cf::DeadSlots::NAME => Some(cf::SlotMeta::slot(cf::SlotMeta::index(key))),
+        cf::DuplicateSlots::NAME => Some(cf::SlotMeta::slot(cf::SlotMeta::index(key))),
+        cf::ErasureMeta::NAME => Some(cf::ErasureMeta::slot(cf::ErasureMeta::index(key))),
+        cf::BankHash::NAME => Some(cf::BankHash::slot(cf::BankHash::index(key))),
+        cf::Root::NAME => Some(cf::Root::slot(cf::Root::index(key))),
+        cf::Index::NAME => Some(cf::Index::slot(cf::Index::index(key))),
+        cf::ShredData::NAME => Some(cf::ShredData::slot(cf::ShredData::index(key))),
+        cf::ShredCode::NAME => Some(cf::ShredCode::slot(cf::ShredCode::index(key))),
+        cf::TransactionStatus::NAME => Some(cf::TransactionStatus::slot(
+            cf::TransactionStatus::index(key),
+        )),
+        cf::AddressSignatures::NAME => Some(cf::AddressSignatures::slot(
+            cf::AddressSignatures::index(key),
+        )),
+        cf::TransactionMemos::NAME => None, // does not implement slot()
+        cf::TransactionStatusIndex::NAME => None, // does not implement slot()
+        cf::Rewards::NAME => Some(cf::Rewards::slot(cf::Rewards::index(key))),
+        cf::Blocktime::NAME => Some(cf::Blocktime::slot(cf::Blocktime::index(key))),
+        cf::PerfSamples::NAME => Some(cf::PerfSamples::slot(cf::PerfSamples::index(key))),
+        cf::BlockHeight::NAME => Some(cf::BlockHeight::slot(cf::BlockHeight::index(key))),
+        cf::ProgramCosts::NAME => None, // does not implement slot()
+        cf::OptimisticSlots::NAME => {
+            Some(cf::OptimisticSlots::slot(cf::OptimisticSlots::index(key)))
+        }
+        &_ => None,
+    }
+}
+
+fn print_blockstore_file_info(
+    blockstore: &Blockstore,
+    file_name: &Option<String>,
+) -> Result<(), String> {
+    match blockstore.live_files_metadata() {
+        Ok(live_files) => {
+            // All files under live_files_metadata are prefixed with "/".
+            let sst_file_name = file_name.as_ref().map(|name| format!("/{}", name));
+            for file in live_files {
```

Comment:
Do these come back in any particular order? I imagine that it will be helpful to view all the files for one column family adjacent to each other, and for each column family, maybe in increasing start key (although things could get goofy with different levels).

Reply:
I think they do (although this behavior is not documented). A better way to do this is via [get_column_family_metadata](https://github.com/rust-rocksdb/rust-rocksdb/pull/657) that I plan to port to rust-rocksdb.

Reply:
Yep, it's sorted by column family:

```
$ ./solana-ledger-tool print-file-info --ledger /mnt/disks/ledger/mainnet/
[/001434.sst] cf_name: meta, level: 0, start_slot: Some(143642862), end_slot: Some(143663961), size: 921933, num_entries: 983
[/001366.sst] cf_name: meta, level: 0, start_slot: Some(143642588), end_slot: Some(143663209), size: 935010, num_entries: 952
[/001284.sst] cf_name: meta, level: 0, start_slot: Some(143642308), end_slot: Some(143662497), size: 856478, num_entries: 910
[/001217.sst] cf_name: meta, level: 1, start_slot: Some(143635799), end_slot: Some(143661855), size: 17901824, num_entries: 24601
[/001435.sst] cf_name: erasure_meta, level: 0, start_slot: Some(143663208), end_slot: Some(143663961), size: 1157783, num_entries: 21389
[/001367.sst] cf_name: erasure_meta, level: 0, start_slot: Some(143662497), end_slot: Some(143663209), size: 1174356, num_entries: 21699
[/001285.sst] cf_name: erasure_meta, level: 0, start_slot: Some(143661853), end_slot: Some(143662497), size: 1105676, num_entries: 20436
```

Reply:
Ok cool, didn't know if the sorting was guaranteed or not. Similar to other commands, if you want to hone in on specific areas, one might pipe and do something like `grep "cf_name: meta"` to get specific cf